hudi-common/src/main/java/org/apache/hudi/common/engine/HoodieReaderContext.java

```diff
@@ -380,21 +380,23 @@ private BiFunction<T, Schema, String> virtualKeyExtractor(String[] recordKeyFiel
   /**
    * Gets the ordering value in particular type.
    *
-   * @param record An option of record.
-   * @param schema The Avro schema of the record.
-   * @param orderingFieldName name of the ordering field
+   * @param record             An option of record.
+   * @param schema             The Avro schema of the record.
+   * @param orderingFieldNames name of the ordering field
    * @return The ordering value.
    */
   public Comparable getOrderingValue(T record,
                                      Schema schema,
-                                     Option<String> orderingFieldName) {
-    if (orderingFieldName.isEmpty()) {
-      return DEFAULT_ORDERING_VALUE;
+                                     Option<List<String>> orderingFieldNames) {
```

Comment:
Would it be simpler to just use an empty list instead of an empty option when there are are no ordering fields specified?

Reply:
+1 with @the-other-tim-brown , let's use empty list and array to represent the non ordering value use cases.